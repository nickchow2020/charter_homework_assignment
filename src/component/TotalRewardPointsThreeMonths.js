import helper from '../helper';
import TotalRewardPointsCell from './TotalRewardPointsCell';

const TotalRewardPointsThreeMonths = ({ totalData }) => {
  const buyers = helper.calculatePointsForEachBuyerId(totalData);
  const buyerIds = helper.getKeysWithUniqueID(buyers);
  return (
    <>
      <h3>Total Three Months:</h3>
      <table style={{ width: '100%', textAlign: 'center' }}>
        <thead>
          <tr>
            <th>BuyerId</th>
            <th>Name</th>
            <th>Total Points</th>
          </tr>
        </thead>
        <tbody>
          {buyerIds.map((obj) => (
            <TotalRewardPointsCell
              key={obj.id}
              totalData={buyers}
              buyerId={obj.key}
            />
          ))}
        </tbody>
      </table>
    </>
  );
};

export default TotalRewardPointsThreeMonths;
