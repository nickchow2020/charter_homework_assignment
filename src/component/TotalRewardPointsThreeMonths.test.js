import { render, screen } from '@testing-library/react';
import TotalRewardPointsThreeMonths from './TotalRewardPointsThreeMonths';

describe('<TotalRewardPointsThreeMonths />', () => {
  const mockData = [
    {
      buyerId: 'tes1',
      rewardPoints: 1,
    },
    {
      buyerId: 'tes3',
      rewardPoints: 2,
    },
    {
      buyerId: 'tes2',
      rewardPoints: 3,
    },

    {
      buyerId: 'tes2',
      rewardPoints: 4,
    },

    {
      buyerId: 'tes3',
      rewardPoints: 5,
    },
    {
      buyerId: 'test3',
      rewardPoints: 6,
    },

    {
      buyerId: 'test3',
      rewardPoints: 7,
    },
  ];

  it('should match snapshot', () => {
    const { asFragment } = render(
      <TotalRewardPointsThreeMonths totalData={mockData} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should contain h3 title', () => {
    render(<TotalRewardPointsThreeMonths totalData={mockData} />);
    const title = screen.getByText('Total Three Months:');
    expect(title).toBeInTheDocument();
  });
});
