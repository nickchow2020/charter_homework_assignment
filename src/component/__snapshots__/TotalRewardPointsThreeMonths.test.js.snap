// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TotalRewardPointsThreeMonths /> should match snapshot 1`] = `
<DocumentFragment>
  <h3>
    Total Three Months:
  </h3>
  <table
    style="width: 100%; text-align: center;"
  >
    <thead>
      <tr>
        <th>
          BuyerId
        </th>
        <th>
          Name
        </th>
        <th>
          Total Points
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <th>
          tes1
        </th>
        <td />
        <td>
          1
        </td>
      </tr>
      <tr>
        <th>
          tes3
        </th>
        <td />
        <td>
          7
        </td>
      </tr>
      <tr>
        <th>
          tes2
        </th>
        <td />
        <td>
          7
        </td>
      </tr>
      <tr>
        <th>
          test3
        </th>
        <td />
        <td>
          13
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
